src/wallet/rpcwallet.cpp

```diff
@@ -1769,19 +1820,64 @@ UniValue listtransactions(const JSONRPCRequest& request)
         return NullUniValue;
     }
 
-    if (request.fHelp || request.params.size() > 4)
-        throw std::runtime_error(
-            "listtransactions ( \"account\" count skip include_watchonly)\n"
+    std::string help_text {};
+    if (!IsDeprecatedRPCEnabled("accounts")) {
+        help_text = "listtransactions (dummy count skip include_watchonly)\n"
+            "\nReturns up to 'count' most recent transactions skipping the first 'from' transactions for account 'account'.\n"
+            "Note that the \"account\" argument and \"otheraccount\" return value have been removed in V0.17. To use this RPC with an \"account\" argument, restart\n"
+            "bitcoind with -deprecatedrpc=accounts\n"
+            "\nArguments:\n"
+            "1. \"dummy\"    (string, optional) If set, should be \"*\" for backwards compatibility.\n"
```

Comment:
Since we're splitting the function anyway, why have the dummy at all?

I also wonder if we should have some form of upgrade plan for when we do this. I know e.g. @luke-jr prefers option objects.

Reply:
I agree, I don't see a reason to support the case "I want the new behavior without accounts but keep the RPC argument". People that wants to avoid breaking change launch with `-deprecatedrpc=accounts`.

Reply:
@promag - please see https://github.com/bitcoin/bitcoin/pull/12953#issuecomment-380813038 for rationale.